api/src/wfl/util.clj

```diff
@@ -388,3 +399,30 @@
    backward or forward depends on the sign of n."
   [^Integer n]
   (.plus (today) n ChronoUnit/DAYS))
+
+(defn randomize
+  "Append a random suffix to `string`."
+  [string]
+  (->> (str/replace (UUID/randomUUID) "-" "") (str string)))
+
+(defn curry
+  "Curry the function `f` such that its arguments may be supplied across two
+   applications."
+  [f]
+  (fn [x & xs] (apply partial f x xs)))
+
+(defn >>>
+  "Left-to-right function composition, ie `(= (>>> f g) (comp g f))`."
+  [f & fs]
+  (reduce #(comp %2 %1) f fs))
+
+(defn poll-while [predicate thunk]
+  (loop [attempt 1]
+    (let [x (thunk)]
+      (if-not (predicate x)
+        x
+        (do (when (< 3 attempt)
+              (throw (TimeoutException. "Max number of attempts exceeded")))
+            (log/debugf "Sleeping - attempt #%s" attempt)
+            (.sleep TimeUnit/MILLISECONDS 100)
+            (recur (inc attempt)))))))
```

Comment:
Because importing the entity then get sometimes returns empty. This code tries at most 3 times to call the get. I've just pulled out the thunk.

Reply:
> Because importing the entity then getting the entity sometimes returns empty.

That is weird… So the point of this function is more like repeat more times than wait?

Reply:
No, the goal is to poll the endpoint until the new entity is visible. It'll call at most 3 times.